### DISCUSSION TITLE
Windows 10 on Arm

### DISCUSSION THREAD

**Discussion by revnode** [Kong/insomnia#3741] [Ideas]
**Is your feature request related to a problem? Please describe.**
Unable to use Insomnia on Windows 10 on Arm.

**Describe the solution you'd like**
A build for Windows 10 on Arm.

**Describe alternatives you've considered**
Curl? I suppose I could use curl.

**Additional context**
https://www.electronjs.org/docs/tutorial/windows-arm

I'm happy to help with this.

**Comment by gschier** [comment]
Ah yes, ARM seems like a good target to support as it's likely the future of computing. 

From that article, it seems like there are a few hurdles to overcome:

- Node 12 is recommended, which we can't use until we upgrade electron #1847
- Will likely need an ARM CI environment to build within (does GH Actions support Windows ARM?)

It seems like it should be fairly trivial after these two things are in place (knock on wood).

**Comment by revnode** [comment]
I was able to build it without issue on a CentOS 8 x64 Intel system running node v10.16.3

```
git clone https://github.com/Kong/insomnia.git
cd insomnia
npm explore npm -g -- npm install node-gyp@latest
yum install libcurl-devel fontconfig-devel
set npm_config_arch=arm64
npm run bootstrap
npm run app-build
```

Of course, it built it for Linux arm64, so not terribly useful unless you're running Linux. Will try to build it on Windows 10 x64 Intel next.

**Comment by revnode** [comment]
Didn't have much luck with Windows 10. Crashed during packaging. I think it's related to electron-builder. It would build for x64_x86, but crash when specifying arm64 as a target.

**Comment by bartdorsey** [comment]
What about building this for RaspberryPi? Seems like it should be similar since it's ARM as well.... Also Apple shipped ARM macs so it seems ARM really should be figured out.

**Comment by LouisStAmour** [comment]
As FYI, I've made a specific ticket to track ARM M1 support over here: #2964

For other types of ARM support, there is this if it's needed: https://github.blog/changelog/2019-12-03-github-actions-self-hosted-runners-on-arm-architectures/ But I imagine cross-compiling, like on macOS for ARM, is also an option these days.
There's some recent interest in getting ARM for Windows builds running under Github Actions: https://github.com/actions/runner/pull/785

Not sure if it helps much, as I'm not familiar with current Electron code, but here's some of what Microsoft did with VS Code to support ARM in various flavours: https://github.com/microsoft/vscode/blob/d1fff1a08360f0d35f124c990dc65a6bccd748d2/build/lib/electron.js#L85 and https://github.com/microsoft/vscode/blob/7d57a8f6f546b5e30027e7cfa87bd834eb5c7bbb/resources/linux/snap/electron-launch#L9-L12 etc. https://github.com/microsoft/vscode/blob/a5f84617e22e6e32afc18a808828f1e233361244/build/gulpfile.vscode.linux.js#L260-L264 for Linux
https://github.com/microsoft/vscode/blob/a5f84617e22e6e32afc18a808828f1e233361244/build/gulpfile.vscode.win32.js#L137-L139 for Windows

... ah, actually the best thing to look for in VS Code's build toolchain is probably `VSCODE_ARCH` as in the number of times it appears in https://github.com/microsoft/vscode/blob/b7aa0f7d406430a388e02b011222cb8e0ea565de/build/azure-pipelines/product-build.yml for their use of Azure Pipelines - https://azure.microsoft.com/en-ca/blog/announcing-azure-pipelines-with-unlimited-ci-cd-minutes-for-open-source/ That said they appear to compile on Linux x64 first and presumably use qemu to run other architectures' build steps on Linux, so they don't need to natively use ARM, it looks like... https://github.com/microsoft/vscode/blob/b7aa0f7d406430a388e02b011222cb8e0ea565de/build/azure-pipelines/product-build.yml#L14-L47

**Comment by JHubi1** [comment]
Any updates for arm64 on Windows 11?

**Reply by subnetmarco** [reply]
@filfreire we already support Apple Silicon, so I think it wouldn't be a stretch to add arm64 on Windows?

**Reply by lgneri** [reply]
Any updates on this topic?

**Reply by jackkav** [reply]
We recently added arm support for linux, so this is technically not far off

**Reply by ravero** [reply]
How far @jackkav?

**Reply by jackkav** [reply]
@ravero I understand your frustration. I added it but it was removed temporarily because arm actions weren't supported in private repos, this is no longer the case, @notjaywu removed it and will put it back once he has bandwidth.

**Comment by ravero** [comment]
Fellows, its been 5 years since this was opened, does it really take that long to include an architecture?

Windows on Arm is a reality for developers, every major vendor of tooling already has a version of this.

**Reply by ravero** [reply]
Its a shame, the app is barely usable on Windows on ARM...

**Comment by Leftyx** [comment]
On my Windows ARM (Surface), Insomnia is barely usable. Time to ditch it. 
It looks like there are alternatives: Bruno or Yaak are both open source and Windows ARM64 native.
